### ISSUE TITLE
Cannot parse identifiers containing surrogate pairs

### ISSUE THREAD

**Issue by zooba** [Microsoft/PTVS#1637]
In Python 3.6's `test_unicode_identifiers.py` file there is an identifier `x\U000E0100`.

This is a valid identifier, however, because our tokenizer operates on `char`s we don't have enough information available to test this.

Each part of the value returns as `UnicodeCategory.Surrogate`. However, if we have a string and pass the start index we get `UnicodeCategory.NonSpacingMark`, which we already treat as a valid character for continuing an identifier.

**Comment by zooba** [comment]
`ParserTests.cs` explicitly skips `test_unicode_identifiers.py` because of this bug.

**Comment by zooba** [comment]
Requires deep changes to the tokenizer, so let's not touch this for 3.0.

**Comment by MikhailArkhipov** [comment]
Is this something commonly used? I know it maybe a standard, but like in many other languages people actually use UTF-8 rather than encoding characters (ex HTML or CSS).

**Comment by int19h** [comment]
I believe it would also repro if you used the actual character in place of the escape - it's just that the escape is easier to spell in the test. Either way, the problem is that C# chars are 16-bit, and we look at strings char by char, so we don't properly handle cases where a Unicode character is used that can't be encoded as a single 16-bit char.

**Comment by zooba** [comment]
Yeah, the problem is that _our_ translation (to UTF-16) cannot be worked around by a user, and it will result in incorrect behavior on valid input. However, as it's such a big change with a small impact, we're not exactly rushing to do it.

**Comment by RaymonGulati1** [comment]
Followup due to internal bug #750355: 

Although certain Unicode characters are valid and able to run on python 3.0+, PTVS shows errors and autocompletions do not work. The file/program will not run (unless error checking before running is disabled) which is incorrect because the python interpreter is able to work with these characters. 

Sample Unicode characters: "test_𠀀𠀁𠀂𠀃𪛑𪛒𪛓"
Error example: 
![image](https://user-images.githubusercontent.com/3315992/52599058-66c23680-2e0c-11e9-987a-6dabb7642748.png)
